server/http_hooks.go

```diff
@@ -45,3 +51,66 @@ func (p *Plugin) handleStatus(w http.ResponseWriter, r *http.Request) {
 func (p *Plugin) handleHello(w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte("Hello World!"))
 }
+
+func (p *Plugin) handleEphemeralUpdate(w http.ResponseWriter, r *http.Request) {
+	request := model.PostActionIntegrationRequestFromJson(r.Body)
+	URL := fmt.Sprintf("%s", *p.API.GetConfig().ServiceSettings.SiteURL)
+
+	count := int(request.Context["count"].(float64)) + 1
+
+	post := &model.Post{
+		Id:        request.PostId,
+		ChannelId: request.ChannelId,
+		UserId:    request.UserId,
+		Message:   fmt.Sprintf("updated ephemeral action %d", count),
+		Props: model.StringInterface{
+			"attachments": []*model.SlackAttachment{
+				{
+					Actions: []*model.PostAction{
+						{
+							Id: model.NewId(),
+							Integration: &model.PostActionIntegration{
+								Context: model.StringInterface{
+									"count": count,
+								},
+								URL: fmt.Sprintf("%s/plugins/%s/ephemeral/update", URL, manifest.Id),
+							},
+							Type: model.POST_ACTION_TYPE_BUTTON,
+							Name: "Update",
+						},
+						{
+							Id: model.NewId(),
+							Integration: &model.PostActionIntegration{
+								Context: model.StringInterface{},
+								URL:     fmt.Sprintf("%s/plugins/%s/ephemeral/delete", URL, manifest.Id),
+							},
+							Type: model.POST_ACTION_TYPE_BUTTON,
+							Name: "Delete",
+						},
+					},
+				},
+			},
+		},
+	}
+	p.API.UpdateEphemeralPost(request.UserId, post)
+
+	resp := &model.PostActionIntegrationResponse{}
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	_, _ = w.Write(resp.ToJson())
+
+}
+
+func (p *Plugin) handleEphemeralDelete(w http.ResponseWriter, r *http.Request) {
+	request := model.PostActionIntegrationRequestFromJson(r.Body)
+
+	post := &model.Post{
+		Id: request.PostId,
+	}
+	p.API.DeleteEphemeralPost(request.UserId, post)
```

Comment:
This is 100% correct. But it looks awful. We have to fix this API method some day...

Reply:
100% agreed.  I noticed you brought it up in ~extensions earlier today, so I look forward to the new version.

Reply:
Well, this got fixed faster then I thought. What do you think about adapting the new API and requiring MM `v5.12.0` for this plugin.

Reply:
Sounds good.  I tried
```	github.com/mattermost/mattermost-server v5.12.0+incompatible``` in go.mod, but there is an error  `unknown revision v5.12.0`.  Leaving as is, and changing `const minimumServerVersion = "5.12.0"`

Reply:
Seems that using minimumServerVersion at 5.12 isn't available yet on master & release-5.12 branches haven't updated version yet.  

Also, when i update go.mod/go.sum to the latest mattermost master, I get these seemingly unrelated errors
```
# github.com/mattermost/mattermost-plugin-demo/server [github.com/mattermost/mattermost-plugin-demo/server.test]
server/configuration.go:189:28: p.Helpers undefined (type *Plugin has no field or method Helpers)
server/configuration_test.go:71:28: undefined: plugintest.Helpers
server/configuration_test.go:86:26: undefined: plugintest.Helpers
server/configuration_test.go:87:17: undefined: plugintest.Helpers
server/configuration_test.go:106:26: undefined: plugintest.Helpers
server/configuration_test.go:107:17: undefined: plugintest.Helpers
server/configuration_test.go:124:26: undefined: plugintest.Helpers
server/configuration_test.go:125:17: undefined: plugintest.Helpers
server/configuration_test.go:145:5: p.SetHelpers undefined (type Plugin has no field or method SetHelpers)
```

What would you like me to do @hanzei ?  Update and fix the changes, or leave this to be fixed in a future PR?  (or other) . happy to do whichever.  :)

(paused changing to the new DeleteEphemeralPost until this is resolved.)

Reply:
You can update the dependency with `go get github.com/mattermost/mattermost-server@2d3e41783398dc16689e5c1732f9d21727383e2c`. Would you please try this?


We can bump `minimumServerVersion` once DB bump is done. This should be done in the next couple of days.

Reply:
That works perfectly @hanzei 👍